lib/cloudregister/registerutils.py

```diff
@@ -1407,6 +1407,19 @@ def write_framework_identifier(cfg):
         framework_file.write(json.dumps(identifier))
 
 
+# ----------------------------------------------------------------------------
+def instance_has_ip_enabled(rmt_ip):
+    """Check if the instance can make IPv4 or IPv6 requests."""
+    try:
+        socket.create_connection((rmt_ip, 443), timeout=2)
```

Comment:
Needs to be assigned to a variable and on success we need to call `close()`. While the temporary socket will be reaped on the Python side the Python garbage collector will not close the socket meaning it will be open on the server side because no request was received by the server. Meaning we will keep connections open unnecessarily and consuming resources on the server for no reason.